packages/sirius-web/backend/sirius-web-application/src/main/java/org/eclipse/sirius/web/application/views/explorer/services/ExplorerNavigationService.java

```diff
@@ -55,29 +54,32 @@ public ExplorerNavigationService(IIdentityService identityService, IRepresentati
     public List<String> getAncestors(IEditingContext editingContext, String treeItemId, Tree tree) {
         List<String> ancestorsIds = new ArrayList<>();
 
-        var optionalRepresentationMetadata = new UUIDParser().parse(treeItemId).flatMap(this.representationDataSearchService::findMetadataById);
-        var optionalSemanticObject = this.getTreeItemObject(editingContext, treeItemId, tree);
+        Optional<Object> optionalObject = this.getParentSemanticObject(treeItemId, ancestorsIds, editingContext, tree);
+        while (optionalObject.isPresent()) {
+            String parentId = this.getItemId(optionalObject.get());
+            ancestorsIds.add(parentId);
+            optionalObject = this.getParentSemanticObject(parentId, ancestorsIds, editingContext, tree);
+        }
+        return ancestorsIds;
+    }
+
+    private Optional<Object> getParentSemanticObject(String elementId, List<String> ancestorsIds, IEditingContext editingContext, Tree tree) {
+        Optional<Object> result = Optional.empty();
+
+        var variableManager = new VariableManager();
+        var optionalRepresentationMetadata = new UUIDParser().parse(elementId).flatMap(this.representationDataSearchService::findMetadataById);
+        var optionalSemanticObject = this.getTreeItemObject(editingContext, elementId, tree);
+        var optionalTreeDesceiption = this.getTreeDescription(editingContext, tree);
 
-        Optional<Object> optionalObject = Optional.empty();
         if (optionalRepresentationMetadata.isPresent()) {
-            // The first parent of a representation item is the item for its targetObject.
-            optionalObject = optionalRepresentationMetadata.map(RepresentationDataMetadataOnly::targetObjectId)
-                    .flatMap(objectId -> this.getTreeItemObject(editingContext, objectId, tree));
-        } else if (optionalSemanticObject.isPresent()) {
-            // The first parent of a semantic object item is the item for its actual container
-            optionalObject = optionalSemanticObject.filter(EObject.class::isInstance)
-                    .map(EObject.class::cast)
-                    .map(eObject -> Optional.<Object> ofNullable(eObject.eContainer()).orElse(eObject.eResource()));
+            variableManager.put("representationDataMetadata", optionalRepresentationMetadata.orElse(null));
```

Comment:
This will not be possible in all use cases. This `parentObjectProvider` API will be used by every tree based representations, as a result it should work in all those use cases. You cannot make it work for those who want to display representations only by having hardcoded a special use case here. The check involving `representationDataSearchService` needs to be done by the `parentObjectProvider`, not here.

Side note, never add a variable conditionally. A variable should always be there even if it's vague is null. Otherwise AQL expression will fail to compile. If the `parentObjectProvider` was written with AQL, it would fail to compile if `self` is not a representation.

Reply:
I removed this specific variable and replaced it with the `treeItemId` which is more general and agnostic. 
So the representation-specific code is now moved to `parentObjectProvider`